### PR TITLE
[SPARK-29656][ML][PYSPARK] ML algs expose aggregationDepth

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
expose expert param `aggregationDepth` in algs: GMM/GLR

### Why are the changes needed?
SVC/LoR/LiR/AFT had exposed expert param aggregationDepth to end users. It should be nice to expose it in similar algs.

### Does this PR introduce any user-facing change?
yes, expose new param

### How was this patch tested?
added pytext tests

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/feature/RobustScaler.scala [hunk 1]
```diff
@@ -34,7 +34,8 @@ import org.apache.spark.sql.types.{StructField, StructType}
 /**
  * Params for [[RobustScaler]] and [[RobustScalerModel]].
  */
-private[feature] trait RobustScalerParams extends Params with HasInputCol with HasOutputCol {
+private[feature] trait RobustScalerParams extends Params with HasInputCol with HasOutputCol
+  with HasAggregationDepth {
```
Comment (huaxingao) [inline mllib/src/main/scala/org/apache/spark/ml/feature/RobustScaler.scala]: I guess you forgot to put ```setAggregationDepth``` in ```class RobustScaler```

Comment (zhengruifeng) [reply mllib/src/main/scala/org/apache/spark/ml/feature/RobustScaler.scala]: Good catch! thanks!

**Comment by zhengruifeng** [comment]
retest this please

**Comment by srowen** [comment]
Hm, I am really not sure users can meaningfully tune this, so I'm not sure about adding so much complexity for it. In some cases I think we use reduce vs treeReduce on purpose as it can be faster; it really depends on the usage.

**Comment by zhengruifeng** [comment]
@srowen Thanks for reviewing!
This PR is for consistency, I will review it again and try to avoding adding to much complexity.

**Comment by zhengruifeng** [comment]
retest this please

**Comment by zhengruifeng** [comment]
merged to master, thanks all

**Comment by srowen** [comment]
Retroactively: OK I understand the argument was consistency. I might have pinged for a confirmation first though as I wasn't in favor of this at first look.

**Comment by zhengruifeng** [comment]
@srowen Sorry for my careless merge. Should I revert it first?
I had revert changes in ALS/Features, now there are 3 main changes:
1, GMM add aggregationDepth;
2, GLR add aggregationDepth, to do this, I need to add this new param in `IterativelyReweightedLeastSquares` && `WeightedLeastSquares`;
3, For the change `reduce` => `treeReduce` in `VectorIndexer`, I left it alone by mistake, I guess I need to revert it in some way.

**Comment by srowen** [comment]
I think it's OK if it's for consistency.

**Pull request closed without merging**